### ISSUE TITLE
Wound Design Document

### ISSUE THREAD

**Issue by KorGgenT** [CleverRaven/Cataclysm-DDA#64424]
### Is your feature request related to a problem? Please describe.

.

### Solution you would like.

.

### Describe alternatives you have considered.

.

### Additional context

Wounds have been a feature the devs have been discussing for a while, and it's high time I lay out some low-level discussion on wounds. We've already talked about high-level topics and general ideas about wounds, but this issue is meant to be a discussion among people who will be using the json and architecting the code.

# What is a wound?
Besides things like "name" and "description" that every other object has, wounds are an object that have a very specific purpose.
- bleeding
a great deal of wounds cause bleeding. this is probably the easiest piece of a wound's json; we can discuss here some regular thresholds that we can use for enums, or i can simply leave it as a number.
- pain
practically all wounds cause pain. it's a pretty simple thing to add to the json, though we could also use an enum to make it easier to create new wounds.
- negative status effects
broken bones are a type of wound that already exists in a fashion, and the effect it would have might be less usage of the limb that's broken. I had gotten the idea the other day to use the enchantment infrastructure to this, and link it to limb data to reduce effectiveness of the limb
- negative effective armor
it hurts a lot when a wound gets wounded. you could potentially get wounded worse if a wound is wounded. this will probably remain a static number, possibly by damage type.
- potential death
if your heart stops working, or your brain stops working, you'll die. we're going to hold off on actually implementing this in the initial implementation, but i feel that it should at least be mentioned in this organizational issue.

# Where is a wound?
each limb has a list of wounds. simple as that.

# How do you heal a wound?
A lot like rimworld, wounds will heal over time, and infection will increase over time. We need to discuss a couple of ways to implement some kind of UI on how to select wounds, and also some way to notate in the json what item can be used on what wound (like a splint on a broken bone, or hemostatic powder on a wound that is bleeding)
Additionally, certain things like bandages can reduce the infection rate for the duration of the effect.

# How do you apply a wound?
currently, there is a damage instance in the code. it contains the amount of damage, and the type of damage, mitigated by armor. What we need to discuss is how exactly we want to determine what damage deals what wound.
Currently, I have an idea for some kind of table lookup, where the damage_instance determines which table to look at (by type) and then picks the wound (by damage, and by limb). Ideas on possible better implementations are welcome.

As you can see by this, there is quite a lot to this project, and there could be quite a bit delegated. I've more or less been talking about it on Discord but as my time has become rather limited in the recent past, i decided on a slightly more permanent form of organization.

**Comment by kevingranade** [comment]
So,  turning a damage instance into a wound. 
There will probably be two different flavors of this over time,  one based on damage type/damage amount and the other based on striking surface/ penetration depth. 

The process I'd suggest is roughly the same though,  we annotate wounds with damage types and ranges,  e.g. "cut damage from 10-15" (and later something similar for striking surface and penetration depth), then filter available wounds based on the properties of the strike.  If we end up with multiple wounds,  we can randomly select one,  possibly with some kind of manual or automatic weighting. 

The damage type and intensity is the base criteria,  but we can also have flags for critical hits or specialty attack types or something else.

Supporting this we could have some validation that checks that there are no gaps so that we don't accidentally create situations where certain damage numbers fail to cause wounds.

**Comment by Venera3** [comment]
A thing I'd definitely include would be wound progression and a chance for existing wounds to get hit - one kick to your stomach probably won't do you in unless you're unlucky, but twelve would probably do the job.

**Comment by fungamer2-2** [comment]
How will damage-over-time effects (such as the Poisoned effect) work with this system?

**Comment by Wyghab** [comment]
Should an already damaged limb be easier to wound? In the same vein as Vener3's comment, getting hit repeatedly in the same body part could make it more susceptible to wounds.

**Comment by KorGgenT** [comment]
so i've had another idea instead of adding another object that is essentially a wrapper for a list of wounds - 
assigning location "values." we've talked about this vaguely in the past about where you would hit a bodypart, but essentially a "location" would be a start value and an end value, and if the hit value is between those two values, it hits that wound.
this would simply be a way to interact with just the whole list of wounds attached to the limbs, and just organize them in such a way as to make this possible. that being said, i think i'm going to walk back the "wound object" stuff i did end up doing, because once i started writing it, i came to this idea and realized the other way of doing it didn't really make that much sense.

**Comment by KorGgenT** [comment]
So. we need to talk about wound treatment - and i'm not talking about an in-depth new system, i'm talking about a system that allows us to treat it similarly to how it was before wounds upheaved everything. 
That being said, and with scope firmly in mind, this is how wounds currently work on my branch (i haven't made a pr yet, but the branch is here https://github.com/KorGgenT/Cataclysm-DDA/tree/wounds )
for those who have a bit of coding knowledge.

A wound_type is the object that is read from json. it has the following members that you can assign in json:
name, description - should be self-evident
size - this is related to the wound treatment stuff. a wound occupies virtual "space," at a "location" and a size. so, a wound of size 5 and at location 20, would overlap with another wound at size 0 and location 25. This essentially makes it so we can have compounding wounds. The current state of "location" is that it is randomly generated upon creating the wound.
pain - how much pain the wound causes. more or less static, though we can discuss how this pain can be mitigated. i am well aware that the human body is able to start igonring a certain amount of pain after a certain amount of time but that should be a later project.
sunder - this is "negative armor" if you try to add a wound that'll overlap with another wound that has sunder, it'll be upgraded.
min/max damage - this is how we determine the particular wound upon taking a hit. combined with sunder.
damage_type - same as above, restricts the wound to a specific damage type
limbs - same as above, restricts the wound to specific body parts.
bleed - the amount of bleeding this wound produces per second (untreated)
heal_time - the amount of time it takes to heal this
heals_into - when this wound heals, it turns into another wound. or not.
infects_into - same as above, but the result of an infection


a "wound" is the name of the object that has a wound_type. it contains any of the mutable data.
age - how old the wound is
location - a number from 0-100 as a representation of where the wound is.
intensity_multiplier - this is how we're going to scale wounds without making a wound for every point of damage. it's generally a scale between x1-x2 from the minimum damage to the maximum damage. i plan on making it so if you take more damage than wounds exist, the intensity multiplier will go above 2 when it's above the max damage.
infection - this is basically the same thing as rimworld. it's a number from 0 - 1 that is a representation of the % it is to fully infected. it also affects healing rate of the wound.
contamination - this is a multiplier on infection rate. i plan on making contamination a part of attacks that are "dirty." i'm not quite sure how to utilize it yet, but it will probably have something to do with monster attacks somehow.


how does the player apply wound care? how does wound care "know" what wounds it can apply to? what kind of ui will it be (ui isn't my forte so not too complex the first time around) are there parts of the wounds themselves i have blatantly missed for a first pass?

**Comment by Shiubiha** [comment]
- I believe location shouldn't be 0-100 for every limb, otherwise a would that has a size of 10 on a torso having a size of 10 in the hands does't make sense, and its going to be even worse if more body parts are added (neck, fingers, lower/upper leg, knee, etc)

- maybe you could add a wound depth, for some types of wounds, like cuts, repeated attacks on same wound making it deeper, after certain depth, it starts to do secondary wounds, like minor bleeding at first, as depth goes on, bleeding intesifies, stabs would be lower size higher depth, cuts the opposite, this way you don't need to have intensity tied to bleeding intensity, and intensity also don't need to be tied to infection chance, so deeper wounds have more infection chance, and bruises starts to damage bones after a certain depth, ultimately breaking the bone

- add a new wound types that doesnt exist in CDDA, "scrap" wounds for when you fall into pits, ledges, car/bike crashes, or when you are thrown by brute zombies, "burn" wounds, high pain and infections chance

**Comment by Shiubiha** [comment]
for treatment:

> for bleeding: when players hand is not being used, player actor keeps pressing the worst bleeding wound, only the worst bleeding wound gets pressed by the players hands, (nowadays its like the player has 6 hands and even arterial bleedings stop after some time of hand pressing), the idea is that when a players gets badly hurt and is bleeding a lot, it keeps pressing the wound while it runs for safety, before starting the wound treatment, maybe use hemostatic powder on the spot before running away, but hemostatic powder appliance should't be a flat -INT number modifiers, otherwise a jugular shot will keep being easily fixed by 3 hemostatic powder/bandage appliance, instead, each treatment for bleeeding has a cap

> hemostatic powders and bandages shouldn't be a straight flat -(INT number) applied to the wound, otherwise 3 hemo-poweder/bandages will keep fixing jugular shots with easy, instead, they should both have a maximum efficiency thats decays as time goes by, so bandages and hemo-powders will need to keep being re-applied, the more volume, the faster they decay

> bandage vs hemostatic powder efficiency varies with depth and size, hemo-powder and player hand-pressure being the most effective for low-area high depth wounds, bandages being the most effective for larger area wounds

> players can equip bandage on hands before applying pressure on bleeding-wound, doing this forces the player to keep pressuring the wound with one hand, only being able to fight with the other hands, equipping on both hands would mean full focus on pressuring the wound, again, only the highest bleeding volume wound gets pressed

> very low bleeding volumes wounds don't get hand-pressed because the actor shouldn't even care about this, this is cool because potentially adding-up over time into iron loss

**Comment by Shiubiha** [comment]
for burning wounds: 

- hemostatic powders are useless, instead of changing bandages, most urgent care would be apply water/clean-water/ sterilized/boilled water (reducing infection chance and contamination), antiseptic further reduces infection chance, bandages then should be applied and re-applied after getting dirty, same idea as above, not changing bandages makes contamination level grow

for bashing wounds:

- no idea, massaging the wound, good health, proper levels of some vitamins, protecting the wound to avoid further damage, I would just suggest that each hit in a bash wound makes the wound deeper and it starts to bleed after a certain depth, or even start to damage bones

**Comment by KorGgenT** [comment]
![image](https://user-images.githubusercontent.com/30208401/231659618-51ce5f5a-3052-4cc6-9ed7-6416fe4a39e3.png)
it obviously needs some info, but i think this is a good starting point!

Shiubiha i haven't made it all the way through your comments yet because of how busy i am this week but it's good stuff and i'm definitely going to use some of it.

**Comment by keampe** [comment]
Is it possible to add a "Treatment" column that shows what the current level of treatment the wound has? So if a wound has been cleaned and disinfected, but the bandages haven't been changed today we can see that at a glance on this UI?

A dirty bandage flag of some kind may be idea as well. Filthy means something specific in CDDA so some other flag might be needed here. This could also be used to create dirty clothes over time - requiring washing or causing rashes but that's another PR. 

   - Shane

**Comment by Shiubiha** [comment]
the problem with flags is that they make something either 0 or 1, IMO the bandage should have an absorption rate based on temperature, humidity ( bandages in rain are less effective, recently made bandages are less effective, recently swimming survivor has wet bandages), external-bleeding volume ( we could have a multiplier for how much of the bleeding is leaking out, internal vs external bleeding ), as well as having some options for bandage sizes, and making a flag will also prevent bandages absorbing contaminants as well

how does the game currently calculates such stuff ( like how many calories you spent, how much you bleed ), is everything calculated once per turn, does the game have a scheduler for some calculations happening every once in a while?

**Comment by DeciusBrutus** [comment]
If the intention is to *replace* hit points with wounds, like DF, there will also have to be different severity of wounds based on what is now HP total, or else the things which currently make HP greater won’t increase the amount of damage a limb can take. 

If the intention is to use wounds to reduce limb HP, the way Rimworld does, then the severity of a wound should also scale with limb HP, to avoid having a superficial scratch destroying the weakest limb, while the strongest will be able to withstand multiple of the maximum wound while still retaining HP. 

In either case, “pain” as a general negative effect on the brain isn’t reasonably modeled as the sum of the pain levels of individual wounds. Something like “maximum pain of any wound, plus a small fraction of all other wounds” for the passive pain caused by all wounds would make more sense. Using a wounded limb might also cause additional pain, but that should be deferred until there’s a way to avoid using a limb (wielding with specific hands, one-handed crafting, limping and hand-crawling as movement options). 

Electrical burns, fire burns, and chemical burns are all subtly different; electrical burns are more of a deep tissue, while fire and chemical burns start out superficial and only affect deeper tissue after prolonged/repeated instances of damage. Washing off an electrical zap will do nothing compared to just stopping further zaps, but washing off a steam or chemical burn immediately will reduce the damage lower than just stepping out of the steam or not getting more caustic chemical on you. Electrified slashing or piercing weapons might also cauterize their wounds and have less bleeding than their piercing or slashing elements would suggest. 

On the subject of bleeding, please recognize that absorbent menstrual products don’t do anything to mitigate blood loss, they just absorb the blood and keep it from making a mess. Bandages and other coagulating methods form a barrier that keeps the blood in the circulatory system, while adsorption products just soak up as much as possible and keep blood in themselves and off of clothing and the ground. 

“Infection” could be one thing, but building a system that allows for different types of infections and spread seems worthwhile; a wound could have staph and/or tetanus, and localized necrosis should be different from a systemic blood infection. Scifi or magic diseases like fungal infection as bloodborn or neurological or merely “systemic” infections could generalize lots of things to the immune system, which could itself be modeled as having a single axis of strength which is depleted by and opposes each infection.

**Comment by keampe** [comment]
Something to consider - as someone who has been seriously injured - is that pain isn't constant, it has spikes mostly related to moving a certain way or bumping the injury but sometimes for no reason. This could be tied to exertion level, for ex. 

As to infection: pretty much every wound that breaks the skin has some level of infection and all injuries have some level of inflammation. However, we have an immune system. the infection currently in CDDA is a very bad one like septic shock or something and is not really appropriate to most injuries. There's a benefit to having at least differing severity of infections, most of which will go away given time. The survivor may not know that and it would be a good idea to take antibiotics as a prophylactic. 

Just some thoughts.
   - Shane

**Comment by I-am-Erk** [comment]
So, wound treatment as a general entity can get extremely complex as we go along. At some point I want to see people debriding tissue from a deep wound with a burn component... But that is not for now.

What I think you need to consider in the short term, when you design how treatments apply to wounds, is that some treatments become part of the wound package so that things may happen to them. Your bandages may become soaked with blood, or stuck to the wound, or filthy with infection or from zombie puke. We don't need to track that now but we should store the "bandaged" concept in such a way that we can do something with it later.

In general we currently model three main categories of wound care:
- hemostasis (stopping bleeding): bandage, tourniquet 
- cleaning (through disinfection only, presently, I think)
- protection: medical tape and gauze, adhesive bandage 

I think hemostasis is presently fairly well modelled but I don't know how it's done or if the current models can be transferred to the old one. I do think that we will probably need to incorporate an element of wound size to the bandage used, so for example an adhesive bandage has a small max size after which you just can't use it. 

We currently don't model wound protection all that much and I don't know if you want to get into it either, but one cool thing is that with how you're doing size and location you can now better model your bandages getting damaged in an attack. I've got thoughts there but I have to save this and try to return

**Comment by Venera3** [comment]
The main point where our hemostasis simulation (and it is the best of all mentioned) falls down is bandages/pressure wizarding away the bleeding, never to return. 

- Both bandages and hemostatic powder should *mitigate* blood loss as long as they are applied, but on repeat damage the wound should start from the same intensity of bleed.
- Tourniquets should apply a hard cap to bleeding on the limb in question, but *don't* change the wounds - in fact, they should also pause the heal timer on all of them. They are a stopgap solution to keep you alive until you get definite care.
- Hand pressure should be the equivalent of bandages for the duration and could leave a `manually compressed` lingering bandage-like effect that similarly gets removed by repeat damage exposing the bleed

Contamination leading to infection would let us differentiate pre-Cataclysm actually sterile / freshly sterilized materials from makeshift stuff better. The contamination level could be set per-monster attack to cut down on the bite special casing.

For cleaning/bandaging I'd probably go with a few levels as activities either from the medical or the % menu, the Consume popup would probably not cut it - something like "first aid / clean and bandage / full service" as levels, runnable per-wound or for all your eligable wounds. Full service would ideally include suturing/splinting, proficiencies permitting, and the exact treatment options (and potentially effects) could probably live in the wound itself, e.g.

 ```treatments: [ 
{ method: bandaging, time_mul: 1.5, bleed_mod: [ 0.9, 0.0 ]}, 
{ method : cleaning, cont_mod: [ 0.7, 0.0], time_mul: 2.0},
{ method: suturing, progress: [ 5, -20 ], bleed_mod: [ 1.1, 0.0 ]}
```
translating to "This wound can be bandaged 1,5 times slower than the base bandaging time, leading to a bleed intensity reduction between 10-100% depending on your success roll, cleaned 2 times slower for a 30-100% contamination reduction or sutured (once) to *increase* wound progression and bleeding for a botched treatment or reduce both on success." First aid would amount to hastily slapping a bandage on it (we can probably abstract away fixation needs, for now), cleaning would make secondary complications less likely and passively increase healing rate if contamination has a negative effect on that, while suturing would be the Actual Treatment *if* you can do it (or get an NPC to do it for you competently, down the line).

**Comment by DeciusBrutus** [comment]
>The idea is that when you're fresh and ready to fight that typical enemies
aren't going to have an easy time hurting you significantly. They're just
going to be landing glancing blows that will sometimes cause minor wounds
like bruises or scratches or abrasions, but then as you get tired and as
those wounds accumulate or in some cases compound such as by a bruise
landing on top of another bruise and causing a more severe bruise or an
open contusion, those same attacks are landing more squarely and causing
more severe hits that start more significantly impairing your fighting
ability. This eventually leads to a wound spiral where if you don't get out
of combat even basic enemies are going to be able to take you down.

Implementing that in a consistent manner is also going to require that enemies' combat prowess, including their ability to resist further damage, degrade with injuries to them. If the same injury system is implemented to enemies, it brings up questions about enemies with physiology remarkably different from human baseline. It might be reasonable to consider robotic frames like bones, and wires like nerves, and hydraulic lines like major blood vessels; but blobs don't have anatomy that would be affected similarly, and building bodies for all of the eldritch horrors and portal creatures is a nontrivial required element of using a robust wound system for enemies in general. In addition, an overhaul of the weakpoint system is necessary to implement equivalent handling of injuries to monsters and humans. 

One likely complication that I see is that a good injury system will over-emphasize tactics of delivering bleeding wounds and then retreating to let the bleeding weaken or finish off the enemy. I think that outcome is not undesirable, since it would not work against enemies with physiology that doesn't suffer from blood loss and would encourage weapons that inflict non-superficial bleeding, disrupting the meta of the fastest knives being dominant. 

On the subject of debridement and other forms of wound cleaning, I think that is a great place to bring the tool quality system into play; needlenose pliers and cotton patches with soap could have 'wound cleaning 1', tweezers or a water flosser with clean water 2, and up to a proper medical toolkit having a wound cleaning of 5. A steel needle and thread might have suturing of 1, cyanoacetate glue a suturing of 3, and a suture kit a suturing of 5, with intermediate values used for approximations or better or worse makeshift tools. 

Making things sterile isn't some difficult industrial process; a pressure cooker can destroy bacteria, viral protein coatings, fungal spores, and even prions in 20 minutes at pressure. Some inorganic chemical contaminants might survive the process, but those largely won't be present in linens getting cut up into bandages. Merely boiling or bleaching typical linens will make them sterile enough, although they might not stay that way after being stuffed in a pocket and submerged in the sewer. 

As for modelling reopening a wound, I think that having a 'bleeding' and a 'clotted' state, where pressure and tourniquets reduce 'bleeding' while applied, but bandages and styptics increase 'clotted' over time, and the blood loss per tick is based on bleeding-clotted. Various things could reduce 'clotted' directly, like aspirin in the system, high exertion, or further damage to the area. 

And from a narrative enabling aspect, I think it's important to ask if we want to prevent the kind of story that has a finale that is a staple of action movies, [where Ahnold gets shot in the shoulder and punched around a bit before using an environmental advantage to perform a reversal, deliver a one-liner, and then get medical attention.](https://www.youtube.com/watch?v=aEBEm4-AMBo). If wounds reduce immediate capacity too much, then there isn't room for a comeback, and if there is too little there isn't much point. Things like adrenaline effects mitigating the effects of pain for only several minutes seem ideal for letting injured characters finish one fight at slightly reduced capacity but not keep fighting forever.

**Comment by kevingranade** [comment]
Enemies don't need wounds because they aren't doing the things that wounds
enable.

Monsters are a dime a dozen and nobody is treating their wounds, they don't
survive long enough for any of this to matter.

Monster types especially can stick with a very stripped down system that
tracks total damage taken plus perhaps some cumulative penalties inflicted
by certain attacks.

**Comment by MNG-cataclysm** [comment]
I'm planning on adding the loss of limbs to the game at some point. KorGgenT referred me here to discuss more about this, so I have a plan laid out on how limb loss in combat would work, so far this is it:
When in combat, damage (calculated after armor negations) that equals or exceeds 20% of the limbs maximum HP has a chance of cutting or tearing it off. Using an EOC condition to check this, that attack then has a base 5% chance to shear off the limb. If it does, apply the proper effects/mutations/etc. to the character and give severe bleeding to the part that limb was attached to (hand -> arm, arm -> torso, so on.). Attacks that do enough damage to reach a higher percent threshold then have a greater chance to remove the limb, I’d say in increments of 2% for every additional 10% of the limbs HP, topping out at a 100% chance to shear off the limb when the damage is 1.5 times that of the limb’s maximum HP. 
This would avoid the previously “game-y” system of a flat 5% when the limb reaches 0 HP and would better reflect what kevingranade said on it; in-combat amputation not being the result of damage acclimation, but a critical strike. This also keeps zombies from poking the limb off for a single point of damage.

How's this sound?

**Comment by I-am-Erk** [comment]
I think that is a decent start, though I don't think eoc is the best language for it. Damage type should influence chance of removal though. That way when we add armour changing damage type it will also be able to naturally affect sever chance. Bludgeon damage should be more likely to break or mangle the limb. It takes truly massive bludgeon damage to remove it entirely. Piercing as well. Bullet less so but still a lot. Mostly it's only slashing damage that should be cutting limbs off at any reasonable frequency.

**Comment by MNG-cataclysm** [comment]
Is there any sort of enemy in the game that can do enough damage to break a limb in a single strike?

**Comment by MNG-cataclysm** [comment]
Besides that, couldn't multiple consecutive hits to the area possibly speed up the process of chopping it off, especially with very sharp weaponry? What if we used an effect for such things, increasing in intensity as the limb gets hit? Or, is this too much like the previous system where consecutive damage would add up?

**Comment by I-am-Erk** [comment]
At some point using the wound system described here we'd be able to make partial amputations increase the chance of a full amputation, but the first pass should be the very basic stuff. On the first pass also, limb removal is essentially just a new way to describe player/npc death.

It doesn't really matter if there are any enemies in the game that can chop your arm off in a single strike, that's not how we balance the game. If it becomes a concern for testing you could always add some kind of late-game blade monster.

**Comment by Brambor** [comment]
What about death (severed limb) by thousand cuts? If you cut enemy five times with an axe, chopping a limb off sounds plausible, like chopping down a tree.

I like the limb accumulating damage even when at 0 HP then for each cutting attack the chance of chopping off increases based on the negative HP (-70 represents arm barely hanging together, even punching the bone would break it).

In new (wounds) system, don't stop accumulating dmg if at threshold. And on every cut attack, after applying wound, if sum of all cut wounds is greater then limb HP, then roll a dice if it gets chopped off. The more wounds the higher the chance.

Or add the probability (P1) from existing damage to P2 of newly received damage. Then weak attacks have chance of chopping off only on damaged limbs. Strong attacks have a low chance on healthy limbs but much higher chance on damaged limbs. Still, only cutting attacks can chop off limbs.

**Comment by I-am-Erk** [comment]
The wound system allows for one wound to be on top of another like I said in my previous comment. However, what you're describing is sort of a useless thing to model... if the enemy is able to hack repeatedly at your limb until it comes off, you're already dead.

**Comment by DeciusBrutus** [comment]
I’m really confused at how anyone could think that a strike count amputate an arm without doing enough damage to render it unusable. 

Either by hitting a joint and cutting through connective tissue, or cutting through the bone in addition to other connective tissue. Or when considering non-bony appendages, cutting through all of the nerves and blood vessels and connective tissue.

**Comment by keampe** [comment]
Multiple wounds to the same limb beyond crippling it, if it doesn't just kill the person, would much more likely mangle the limb such that it's permanently disabled or, more likely, have to be amputated to avoid gangrene. This is amputation by a surgeon (ideally) done as part of treatment, not combat. It is highly, highly unlikely for someone to be hit in exactly the same spot over and over each blow whittling the arm away. You're hitting the limb in different locations as a rule.

   - Shane

**Comment by Spicyshadow** [comment]
After reading the discussion on the wound systeem I´m surprised that hygiene and cleanliness never came up. I´m not just talking about your characters washing every few days to keep general grime and dirt off of them although that should have an direct if minor impact on rate of wound contamination and an indirectly impact by boosting health (healing rate) and immunity.

Your character is going to be very filthy very often. Many activities such as butchering, hacking/pulping corpses and dissecting are going to have your character covered in blood, gore, etc. Your character is probably also going to be plastered in body fluids if they hack their way though dozens of enemies or get covered in bile by a boomer. This level of filthy would very quick contaminate any wound with very infectious material. Cleaning the zombie guts off of yourself and your clothing should be pretty high on the piority list if you don´t want your cleaned and bandaged wounds to get infected.

**Comment by KorGgenT** [comment]
i'd appreciate we keep on the topic of wounds in this design document. that is a very long and well-tread topic that i do not want to bog down this issue with.

**Comment by zachary-kaelan** [comment]
> opens path to finally do drugs the right way, a drug is just an invisible wound (0 depth, 0 area, 0 intensity, 0/low infection chance) with a contaminant that enters the bloodstream, so a heroin shot is a contaminant that enters bloodstream virtually in an instant, alcohol is a contaminant that enters bloodstream slowly, etc... drugs can be non-pure too, with unknown mixed drugs (other contaminants), varied rates of purity, random dirty, etc

What you're describing is vitamins. I was actually already in the process of adding a vitamin for eye contamination to model conjunctivitis from extended smoke exposure, and of adding a filth vitamin accumulated from eating tainted materials and wearing filthy clothing.

I propose two new features, as infrastructure for wounds and their effects:
* A new type of "vitamin", or a new type that's similar to vitamins, for application to specific body parts and/or wounds instead of the body on a systemic level. If too much of the vitamin accumulates in a limb, the effects are automatically applied to that specific limb.
  * This would allow for modeling contamination, chemical burns, venoms, infections, etc. 
  * We could trash the existing systems for tracking dampness of limbs, temperature of limbs, etc.
  * This would allow for properly modeling tourniquets, as we could track the blood loss in that limb.
* I've seen a number of hacks, verbosity, and inconsistencies in the code and JSON when it comes to effects applied to limbs, and I've ran into multiple bugs because of it. Limb-specific effects should be separated into their own type, whether that be a type of wound or something else, having a list of body parts that it can be applied to, and/or a default.
  * Example weirdness:
    * Boomers apply their effect to the eyes in the JSON but in the unit test it isn't.
    * Tear gas is one effect but smoke_eyes and smoke_lungs are separated.
    * There are a long list of different effects that are the same with just a body part changed in the ID, name, and description, 
      * For example, 5 different kinds of winded.
    * If you specify that an effect is applied to a random limb, there's no specifying what body parts it can or cannot be applied to.
  * Alternatively, new fields could be added to the existing effect object, but that wouldn't enforce compatibility with the limb system for any new effects people added.
  * If only one body part is in the list, it should apply to it and check for it there by default. 
  * If multiple body parts are on the list:
    * Then specifying a `RANDOM` body part selects from that list.
    * Specifying an `ALL` body part applies it to everything on the list.
    * If there's no default, specifying no body part throws an error.

On overlapping wounds:
  * There is no way someone is getting two stab wounds in the same exact spot. I've seen numerous murder reports that count 30-50+ individual stab wounds. 
  * If someone gets two intersecting cuts, they will only intersect at a single small point. A second cut isn't going to be at the same angle and same length. Again, murder reports can also count the exact number of individual cuts.
  * Axes *can* hit the same spot twice, but it's still difficult to expand an existing wound unless they're holding still.
    * In CDDA they deal heavy bash damage as well, which is a factor that could be used to differentiate.
  * There's a [TVTropes page](https://tvtropes.org/pmwiki/pmwiki.php/Main/SplittingTheArrow) for hitting the same spot twice with a bullet or arrow.
  * Impact wounds and burns, of course, can definitely accumulate.
  * I'm not convinced that a heat burn would make other wounds worse or vice versa, beyond a flare of pain.
     * In fact, if you heat burn an open wound, it should cauterize it and stop the bleeding.
  * Chemical burns unwashed, however, could definitely worsen or be worsened by other damage types, with spread of the chemicals.
